### DISCUSSION TITLE
Welcome to OpenF1 Discussions!

### DISCUSSION THREAD

**Discussion by br-g** [br-g/openf1#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect and collaborate around the OpenF1 project. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by SUYASH-a17** [comment]
We are working on a project that requires data from the year 2022. Could you kindly assist us in obtaining this data or guide us to a reliable source where we might access it?

**Reply by br-g** [reply]
Hello,

Here is how you can do it:
1. Install the project locally (https://github.com/br-g/openf1/tree/main?tab=readme-ov-file#running-the-project-locally)
2. Ingest season 2022 (https://github.com/br-g/openf1/tree/main/src/openf1/services/ingestor_livetiming/historical#ingest-season)
3. Run the API locally (https://github.com/br-g/openf1/blob/main/src/openf1/services/query_api/README.md#running-the-api-locally)
4. Query the data you are interested in.

I haven't tested ingesting old seasons yet though, so there is no guarantee this will work out of the box.
Cheers

**Reply by gmalbert** [reply]
@br-g It appears that 2019, 2020 both work using this method. 2018 and 2022 get errors (a lot of them). I don't know what's happening on either of those years, whether it's a formatting issue or what. 

On 2021, I get the following error:

`line 110, in process_message
    for driver_number, data in message.content["Lines"].items():
                               ~~~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'Lines'`

Happy to provide logs if you can lead me to those, or I can copy/paste the errors for 2018 and 2022.

@SUYASH-a17 did you have better luck? If so, do tell!
